modules/core/src/main/java/org/apache/ignite/internal/ThinProtocolFeature.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.EnumSet;
+
+/**
+ * The base feature class.
+ */
+public interface ThinProtocolFeature {
+    /**
+     * @return Feature ID.
+     */
+    int featureId();
```

Comment:
"public" modificator must be present on all interface methods (codestyle). But I think the whole interface should be removed (see my comment on ticket for details)